File: apps/frontend/src/adapters/shared/reconciliation.ts
import type {
  ReconciliationConfig,
  ReconciliationResult,
  ResolveRequest,
  ScanResult,
  ImportRun,
} from "@/lib/types";
import { invoke } from "./platform";

export const reconciliationScan = async (): Promise<ScanResult> => {
  return invoke<ScanResult>("reconciliation_scan");
};

export const getReconciliationPending = async (): Promise<ReconciliationResult[]> => {
  return invoke<ReconciliationResult[]>("reconciliation_pending");
};

export const getReconciliationDetail = async (runId: string): Promise<ReconciliationResult> => {
  return invoke<ReconciliationResult>("reconciliation_detail", { runId });
};

export const reconciliationResolve = async (request: ResolveRequest): Promise<ImportRun> => {
  return invoke<ImportRun>("reconciliation_resolve", { request });
};

export const getReconciliationConfig = async (): Promise<ReconciliationConfig> => {
  return invoke<ReconciliationConfig>("get_reconciliation_config");
};

export const updateReconciliationConfig = async (
  config: ReconciliationConfig,
): Promise<ReconciliationConfig> => {
  return invoke<ReconciliationConfig>("update_reconciliation_config", { config });
};

File: apps/tauri/src/commands/reconciliation.rs
fn unsupported<T>() -> Result<T, String> {
    Err("Statement reconciliation is not available in desktop mode.".to_string())
}

#[tauri::command]
pub async fn reconciliation_scan() -> Result<serde_json::Value, String> {
    unsupported()
}

#[tauri::command]
pub async fn reconciliation_pending() -> Result<Vec<serde_json::Value>, String> {
    unsupported()
}

#[tauri::command]
pub async fn reconciliation_detail(run_id: String) -> Result<serde_json::Value, String> {
    let _ = run_id;
    unsupported()
}

#[tauri::command]
pub async fn reconciliation_resolve(
    request: serde_json::Value,
) -> Result<serde_json::Value, String> {
    let _ = request;
    unsupported()
}

#[tauri::command]
pub async fn get_reconciliation_config() -> Result<serde_json::Value, String> {
    unsupported()
}

#[tauri::command]
pub async fn update_reconciliation_config(
    config: serde_json::Value,
) -> Result<serde_json::Value, String> {
    let _ = config;
    unsupported()
}
